polymer/src/amp-my-embed.html

```diff
@@ -0,0 +1,285 @@
+        font-weight: bold;
+      }
+
+      .description {
+        color: var(--app-secondary-color);
+      }
+
+      @media (max-width: 767px) {
+        :host {
+          margin: 0 12px;
+        }
+      }
+
+    </style>
+
+    <span id="name"></span>
+
+  </template>
+
+  <script>
+
+    class AmpMyEmbed extends Polymer.Element {
+
+      static get is() { return 'amp-my-embed'; }
+
+      static get properties() { return {
+
+        url: {
+          type: String,
+          value: window.location.origin
+        },
+        fullUrl: {
+          type: String,
+          computed: 'computeFullUrl(itemUrl)',
+          observer: 'urlChanged'
+        },
+        itemUrl: {
+          type: String,
+          notify: true
+        },
+        content: {
+          type: String,
+          notify: true
+        }
+      };
+    }
+
+      computeFullUrl(itemUrl) {
+        return [this.url, this.itemUrl, "embed"].join('')
+      }
+
+      urlChanged() {
+        this.fetchAndAttachAmpDoc_(this.fullUrl);
+      }
+
+      constructor() {
+        super();
+        // 'offline' is set to true if and when the document fetch fails.
+        this.state = {'offline': false};
+
+        /**
+         * `window.AMP` is set by the AMP runtime when it finishes loading.
+         * @const
+         * @private
+         */
+        this.ampReadyPromise_ = new Promise(resolve => {
+          (window.AMP = window.AMP || []).push(resolve);
+        });
+
+        /**
+         * XMLHTTPRequest that fetches the AMP document.
+         * @private
+         * @type {XMLHTTPRequest}
+         */
+        this.xhr_ = null;
+
+        /**
+         * Provides AMP functionality on the newly created shadow root after
+         * an AMP document is attached.
+         * @private
+         * @type {Object}
+         */
+        this.shadowAmp_ = null;
+
+        /**
+         * The root node of the shadow AMP.
+         * @note A single node must not be reused for multiple shadow AMP docs.
+         * @type {Element}
+         */
+        this.shadowRoot_ = null;
+
+        /** @private */
+        this.boundClickListener_ = this.clickListener_.bind(this);
+
+        /**
+         * Child element that will wrap the AMP shadow root.
+         * @private
+         * @type {Element}
+         */
+        this.container_ = document.getElementById('container');
+
+      }
+
+      ready() {
+        super.ready();
+        this.$.name.textContent = this.shadowAmp_;
+      }
+
+  /**
+   * Fetches the AMP document at `url` and attaches it as a shadow root.
+   * @private
+   * @param {string} url
+   */
+  fetchAndAttachAmpDoc_(url) {
+    this.fetchDocument_(url).then(doc => {
+      return this.ampReadyPromise_.then(amp => {
+        // Hide navigational and other unwanted elements before displaying.
+        this.hideUnwantedElementsOnDocument_(doc);
+
+        // Replace the old shadow root with a new div element.
+        const oldShadowRoot = this.shadowRoot_;
```

Comment:
Is it intended to use `shadowRoot_` instead of the `shadowRoot` that Polymer defines?

Reply:
which is the 

> shadowRoot that Polymer defines?

Reply:
https://www.polymer-project.org/2.0/docs/devguide/shadow-dom

> Polymer provides a declarative mechanism for adding a shadow tree using a DOM template. When you provide a DOM template for an element, Polymer attaches a shadow root for each instance of the element and copies the template contents into the shadow tree.

The `this.shadowRoot` of an element is the reference to the shadow root.

Reply:
I renamed the variable as it was confusing. PTAL.